### PR TITLE
workloopR tech note

### PR DESCRIPTION
**Pull request by vbaliga**
Hi @stefaniebutland, as promised here's a draft of the tech note for the workloopR package (please [see here](https://github.com/ropensci/software-review/issues/326#issuecomment-548498242) for context). I'm happy to follow up as needed

### CODE REVIEW

**Comment by vbaliga** [comment]
Hi again @stefaniebutland, sorry for the false starts. I think it's ready for your review & feedback now. 

I just took a look at the site preview ([url here](https://5dc72756d8c1460008c6a04f--ropensci.netlify.com/technotes/2019/11/14/workloopr-release/)) and I'm not certain what the best way to handle images will be. There are two things it would be great to get your advice on:

1.  Is there a way to center the workloopR hex image (the first image in the post)? I'm not familiar enough with how this can be handled in Hugo
2. It looks like the two work loop images later on are stretched to fit the dimensions of the page. Again sorry for my lack of experience with rendering things in Hugo -- is there a way to set their dimensions specifically so that they appear smaller and side-by-side?

Thanks!!

**Comment by stefaniebutland** [comment]
Thank you @vbaliga. I'm away from work today. Will review Tuesday and get some solutions for images!

**Comment by stefaniebutland** [comment]
Hi @vbaliga ,

**Images**
To limit the size of an image, try setting `width="300"` or your preferred width

Tip: instead of `` ,
using Hugo shortcodes, you can add an image this way:
`{{< figure src="/img/blog-images/2019-11-14-workloopR-release/work_loop_plots-1.png" alt="alt text here" width="300" >}}` and same for the other image.

@maelle how would Vikram center his hex sticker image? (For the record, Vikram I think it looks good where it is)

**Review**
I like how the intro paragraph sets the stage, and how you link to your docs for more info. They complement the tech note really well. All my comments are here

- don't put package names in backticks or bold. No special formatting
- you could note somewhere (e.g. last line of post?) that it's been published in JOSS 🎉 and add to references 
- "typical researcher’s workflow" -> "typical muscle physiology researcher’s workflow"
- "integrating EMG data" -> spell out? and link to explanation fo what it is, especially in case someone wants to investigate further.
- I think you need to give a short context explanation near the start so that a reader knows what the later plots of position vs force are showing, and what the stimulus protocol refers to. 
- Related: In "What it does", when you refer to "variety of features" in the bullet list, I wondered what the terms meant. Can you refer to / link to a resource where curious readers can get more info? Some are covered in your docs overview, but not all I think. 

Happy to take a look at edits when you're ready

**Comment by vbaliga** [comment]
Hi @stefaniebutland 

Thank you very much! I made edits to the tech note following your feedback and I think it's ready for your eyes. 

Thanks for the tips re: images; I hadn't thought to simply use the Hugo shortcodes for the two work loop images that came later on. 

With respect to the hex sticker image -- I've actually grown to like it where it is. So thank you for investigating, but I'll retract that query. 

I was generally happy to incorporate the changes you requested. But I would like to follow up on a couple of your comments, as I am not completely sure I did justice to them:
>  * I think you need to give a short context explanation near the start so that a reader knows what the later plots of position vs force are showing, and what the stimulus protocol refers to.

I did this in two places; one in the "features" bullet list and one in more depth right before the plots. I decided to be more explicit in the latter case so that the description of what a work loop plot is could be fresh in a reader's mind as they see the plots. Do you think these changes fit what you had in mind?

>  * Related: In "What it does", when you refer to "variety of features" in the bullet list, I wondered what the terms meant. Can you refer to / link to a resource where curious readers can get more info? Some are covered in your docs overview, but not all I think.

Happy to do this! I hope I did not go overboard with the hyperlinking I added in. Please let me know if you'd prefer this done another way.

Thanks again for all your feedback & advice!

File: content/technotes/2019-11-14-workloopR-release.Rmd [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+Should you like to suggest a specific feature, please use the [Issues page](https://github.com/ropensci/workloopR/issues/new/choose) of our repository. 
+
+
+### Package review by rOpenSci and JOSS
+
+We are thankful for the suggestions we've already received. workloopR benefited a lot from peer review of code through rOpenSci. Special thanks to Julia Romanowska ([jromanowska](https://github.com/jromanowska)) and Eric Brown ([eebrown](https://github.com/eebrown)) for reviewing our code and giving helpful suggestions on how to improve the clarity of workloopR's presentation.
+
+We are also happy to share that a journal article that accompanies this package was also peer reviewed and accepted by the [Journal of Open Source Software](https://joss.theoj.org/) (JOSS); our paper can be found [here](https://doi.org/10.21105/joss.01856).
```
Comment (stefaniebutland) [inline content/technotes/2019-11-14-workloopR-release.Rmd]:
```suggestion
We are also happy to share that a journal article that accompanies this package was also peer reviewed and accepted by the Journal of Open Source Software (JOSS)[^2].
```

turned paper link to citation format

File: content/technotes/2019-11-14-workloopR-release.Rmd [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+🐢
+
+
+[^1]: Software from Aurora Scientific: https://aurorascientific.com/products/muscle-physiology/muscle-software/
```
Comment (stefaniebutland) [inline content/technotes/2019-11-14-workloopR-release.Rmd]:
```suggestion
[^1]: Software from Aurora Scientific: https://aurorascientific.com/products/muscle-physiology/muscle-software/
[^2]: Baliga V. and Senthivasan S (2019). workloopR: Analysis of work loops and other data from muscle physiology experiments in R. Journal of Open Source Software, 4(43), 1856, https://doi.org/10.21105/joss.01856
```

File: content/technotes/2019-11-14-workloopR-release.Rmd [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+wl_batch_summarized
+
+```
+
+For more, see our [Batch processing vignette](https://docs.ropensci.org/workloopR/articles/batch-processing.html).
+
+
+### How to get workloopR
+
+We are not (yet) on CRAN but the package is available through rOpenSci's github:
```
Comment (stefaniebutland) [inline content/technotes/2019-11-14-workloopR-release.Rmd]:
```suggestion
We are not (yet) on CRAN but the package is available through rOpenSci's GitHub:
```
GitHub capitalization

**Comment by stefaniebutland** [comment]
Your explanations are FANTASTIC! Clear language that gives just the right amount of context with links to other documentation. Thank you for those. 


I see you've done these :-)
- [ ] I have made 2 "GitHub commit suggestions" to add your JOSS citation as a reference. If you agree, you can directly commit.

- [ ] Made 1 GitHub commit suggestion to edit capitalization of GitHub.

After those edits, this is definitely ready for me to publish and tweet on Thursday. Thanks so much for writing this and such nice documentation.

**Comment by vbaliga** [comment]
> I see you've done these :-)
> 
>  * [x]  I have made 2 "GitHub commit suggestions" to add your JOSS citation as a reference. If you agree, you can directly commit.
> 
>  * [x]  Made 1 GitHub commit suggestion to edit capitalization of GitHub.

yup!

> After those edits, this is definitely ready for me to publish and tweet on Thursday. Thanks so much for writing this and such nice documentation.

Sounds great - thank you for all your help!! And thanks again for the opportunity!

**Comment by stefaniebutland** [comment]
Asides:
1. Did you use the [Rmd template](https://github.com/ropensci/rodev/blob/master/inst/rmarkdown/templates/ropensci-blog-post/skeleton/skeleton.Rmd) to start your post? Was it helpful? Anything missing?

2. I'm very happy to see a UBC submission. I'm in Kamloops BC, but worked at UBC in life sciences for *many* years

File: content/technotes/2019-11-14-workloopR-release.Rmd [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+workloopR, as the name implies, was originally designed to handle data from work loop experiments as well as experiments that are complementary to work loops like twitch and tetanic trials. 
+
+We are happy to expand the scope of the package to incorporate even more types of analyses of muscle physiology or biomechanics. We're presently eager to add data import functions for non-`ddf` file types. We're also interested in integrating support for electromyographic ([EMG](https://en.wikipedia.org/wiki/Electromyography)) data recorded directly from muscles, but that may take a little longer to develop.
+
+Should you like to suggest a specific feature, please use the [Issues page](https://github.com/ropensci/workloopR/issues/new/choose) of our repository. 
+
+
+### Package review by rOpenSci and JOSS
+
+We are thankful for the suggestions we've already received. workloopR benefited a lot from peer review of code through rOpenSci. Special thanks to Julia Romanowska ([jromanowska](https://github.com/jromanowska)) and Eric Brown ([eebrown](https://github.com/eebrown)) for reviewing our code and giving helpful suggestions on how to improve the clarity of workloopR's presentation.
```
Comment (stefaniebutland) [inline content/technotes/2019-11-14-workloopR-release.Rmd]:
```suggestion
We are thankful for the suggestions we've already received. workloopR benefited a lot from [peer review of code](https://github.com/ropensci/software-review/issues/326) through rOpenSci. Special thanks to Julia Romanowska ([jromanowska](https://github.com/jromanowska)) and Eric Brown ([eebrown](https://github.com/eebrown)) for reviewing our code and giving helpful suggestions on how to improve the clarity of workloopR's presentation.
```
link to review thread

**Comment by vbaliga** [comment]
>     1. Did you use the [Rmd template](https://github.com/ropensci/rodev/blob/master/inst/rmarkdown/templates/ropensci-blog-post/skeleton/skeleton.Rmd) to start your post? Was it helpful? Anything missing?

I did use the template to get going. Generally very helpful, especially for giving the overall skeleton and header material. The one tweak I might suggest is that the [chunk on line 48](https://github.com/ropensci/rodev/blob/0e56337e6f72014af5a153449d989b867983eb41/inst/rmarkdown/templates/ropensci-blog-post/skeleton/skeleton.Rmd#L48) somehow be edited so that Hugo code for including that figure can also be automated (if desired). Not sure if this is currently feasible. In any case, once I got the hang of things it wasn't all that difficult to incorporate Hugo code; it was more of a matter of realizing that I should use Hugo rather than standard markdown image linking.
 
>     2. I'm very happy to see a UBC submission. I'm in Kamloops BC, but worked at UBC in life sciences for _many_ years

I noticed you're in Kamloops! I'm a relatively new UBC transplant (since 2017, originally from California) but liking it a lot so far. Seems like BC is a great place to be for promoting reproducible research. Thanks again for all your help

**Comment by stefaniebutland** [comment]
@vbaliga Sorry. Two more things
- I added suggested commit to link to the package review
- the changes you committed appear in the Rmd file, but not in the md so they don't show up in the rendered post https://deploy-preview-551--ropensci.netlify.com/technotes/2019/11/14/workloopr-release/

**Comment by vbaliga** [comment]
@stefaniebutland 
No problem! I just reknitted to regenerate the .md file. I think it has all the changes we made today. One thing I noticed is that there is an "Error Embedding" at the [end of the deployed preview](https://deploy-preview-551--ropensci.netlify.com/technotes/2019/11/14/workloopr-release/). Is there an issue in the formatting of the Rmd/md? Or does this get resolved once the site goes live?

**Comment by stefaniebutland** [comment]
> does this get resolved once the site goes live?

Yes. When I add the topicid before publishing, the comments are enabled there.

All good now. Thank you again

**Comment by maelle** [comment]
:wave: @vbaliga! Thanks for the feedback!

> The one tweak I might suggest is that the chunk on line 48 somehow be edited so that Hugo code for including that figure can also be automated (if desired). Not sure if this is currently feasible.

Can you please clarify? When knitted I thought this included the figure? Without Hugo code, that's true, but it should work without it? (and then _I_ will clarify the docs :grin:)

**Comment by maelle** [comment]
The template now lives [in the roblog package](https://github.com/ropenscilabs/roblog/blob/master/inst/rmarkdown/templates/ropensci-blog-post/skeleton/skeleton.Rmd) and I tried clarifying
* plot: use R Markdown chunk options
* not plot: use Hugo shortcode.

**Comment by vbaliga** [comment]
Hi @maelle 

Thanks for taking my feedback. The issue wasn't regarding whether the image was included. Rather, it was that when the .Rmd knit, the syntax within chunks that would customize images (e.g. `fig.height=4, fig.width=4, fig.show="hold"`) was ultimately unused by the deployed Netlify site. Instead, I needed to use Hugo code to specify how images should be presented, which I had not realized at first.

It's great that lines 48 - 50 show an example of how a chunk can still be used to produce an image via R code, but I'd recommend that [line 52](https://github.com/ropenscilabs/roblog/blob/343913f11f2539fefcbad6789f299036e5e0c862/inst/rmarkdown/templates/ropensci-blog-post/skeleton/skeleton.Rmd#L52) be edited to say that if a user wants additional customization of image dimensions and placement...etc, they should replace the markdown image linking (which is produced when the .Rmd is knit) with Hugo image linking. Even better would be to have the .Rmd produce only Hugo code for images instead of the usual `` when it is knit, but I understand that this would take a bit more work! 

Thanks!!

**Comment by maelle** [comment]
Thank you, this makes sense!

**Pull request merged**